### PR TITLE
Update prevalence of env vars to set config and cache paths

### PR DESCRIPTION
**Pull request by finefoot**
See #4894

### CODE REVIEW

File: docs/usage_general.rst.inc [hunk 1]
```diff
@@ -249,22 +249,31 @@ Some automatic "answerers" (if set, they automatically answer confirmation quest
...
     BORG_BASE_DIR
-        Default to '$HOME', '~$USER', '~' (in that order)'.
-        If we refer to ~ below, we in fact mean BORG_BASE_DIR.
+        Defaults to ``$HOME`` or ``~$USER`` or ``~`` (in that order).
+        If you want to move all borg-specific folders to a custom path at once, all you need to do is
+        to modify ``BORG_BASE_DIR``: the other paths for cache, config etc. will adapt accordingly.
+        (Assuming you didn't set them to a different custom value.)
     BORG_CACHE_DIR
-        Default to '~/.cache/borg'. This directory contains the local cache and might need a lot
+        Defaults to ``BORG_BASE_DIR/.cache/borg``. If ``BORG_BASE_DIR`` is not explicitly set while
```
Comment (ThomasWaldmann) [inline docs/usage_general.rst.inc]: as we talk about $HOME, guess we need $BORG_BASE_DIR here when talking about pathes.

Comment (ThomasWaldmann) [reply docs/usage_general.rst.inc]: same at other misc. places, please fix all places within a path to use `$BORG.../...` or `$XDG.../...`.

Comment (finefoot) [reply docs/usage_general.rst.inc]: :+1:

**Comment by finefoot** [comment]
@ThomasWaldmann Friendly ping - isn't https://github.com/borgbackup/borg/pull/4951/commits/f6ba7a343a365a9c8cac9b50decf7787061c7d90 what you meant?

**Comment by finefoot** [comment]
Just asking because in the meantime (after https://github.com/borgbackup/borg/commit/f6ba7a343a365a9c8cac9b50decf7787061c7d90), you merged another pull request changing the same file.

File: docs/usage_general.rst.inc (hunk 1 above)
Comment (ThomasWaldmann) [reply docs/usage_general.rst.inc]: ok, solved.

**Comment by ThomasWaldmann** [comment]
Fine now, but needs conflict resolution, preferably via `git rebase -i master` onto a current master.

Can you do that please?

**Comment by finefoot** [comment]
> Fine now, but needs conflict resolution, preferably via git rebase -i master onto a current master.

@ThomasWaldmann  Yes, sure, I was just asking to get confirmation because I was confused due to you not merging when it was ready. Otherwise if I'd do the rebase and you'll merge another pull requests again, I'd just have to rebase again...

Anyway, conflicts fixed. For now. :wink:

**Comment by finefoot** [comment]
> Otherwise if I'd do the rebase and you'll merge another pull requests again, I'd just have to rebase again...

@ThomasWaldmann Umm, I just noticed that exactly this *has* happened again because the pull request wasn't merged last time I fixed the conflicts. So now, twice it was ready to merge but two times other pull requests introduced merge conflicts afterwards.

You do have push access to the branch, though. So maybe you can just do the final git rebase yourself, immediately before you have time to merge?

**Comment by ThomasWaldmann** [comment]
superseded by #5066.

**Pull request closed without merging**